app/classifier/tasks/shortcut-editor.jsx

```diff
@@ -0,0 +1,136 @@
+import React from 'react';
+import AutoSave from '../../components/auto-save';
+import handleInputChange from '../../lib/handle-input-change';
+import Shortcut from './shortcut';
+
+export default class ShortcutEditor extends React.Component {
+  constructor(props) {
+    super(props);
+    this.addAnswer = this.addAnswer.bind(this);
+    this.toggleShortcut = this.toggleShortcut.bind(this);
+  }
+
+  toggleShortcut(e) {
+    let nextTaskID;
+
+    if (e.target.checked) {
+      const taskCount = Object.keys(this.props.workflow.tasks).length;
+
+      let taskIDNumber = -1;
+
+      while (!((nextTaskID != null) && !(nextTaskID in this.props.workflow.tasks))) {
+        taskIDNumber += 1;
+        nextTaskID = `T${taskCount + taskIDNumber}`;
```

Comment:
Ok, could you make a note about DRYing this up in the classifier rewrite issue #3527? Let's tackle it again later.